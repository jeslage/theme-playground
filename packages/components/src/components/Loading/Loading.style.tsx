import styled from 'styled-components';

interface LoadingProps {
  readonly isLoading: boolean;
}

const StyledLoading = styled.div<LoadingProps>`
  pointer-events: none;
  display: ${props => (props.isLoading ? 'inline-flex' : 'none')};
  visibility: ${props => (props.isLoading ? 'visible' : 'hidden')};
  align-items: center;
  padding: 0.75rem 1rem;
  border-radius: 3rem;
  flex-grow: 2;
  margin-right: 1rem;
  background-color: ${({ theme }) => theme.backgroundDark};
  box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.1);

  div {
    display: block;
    width: 12px;
    height: 12px;
    margin: 0;
    border-radius: 50%;
    border: 2px solid ${({ theme }) => theme.text};
    border-color: ${({ theme }) => theme.text} transparent
      ${({ theme }) => theme.text} transparent;
    animation: loading 1.2s linear infinite;
  }

  span {
    text-transform: uppercase;
    font-weight: bold;
    margin-left: 0.5rem;
  }

  @keyframes loading {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;
export default StyledLoading;
