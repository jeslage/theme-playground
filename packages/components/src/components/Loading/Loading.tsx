import * as React from 'react';

import StyledLoading from './Loading.style';

const Loading: React.FC<{ isLoading?: boolean }> = ({ isLoading }) => {
  return (
    <StyledLoading isLoading={isLoading}>
      <div />
      <span>Updating</span>
    </StyledLoading>
  );
};

export default Loading;
