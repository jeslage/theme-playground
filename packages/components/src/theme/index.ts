export const darkTheme = {
  base: 'dark',
  backgroundDark: '#2f2f2f',
  backgroundLight: '#333',
  border: '#999999',
  text: '#FFFFFF'
};

export const lightTheme = {
  base: 'light',
  backgroundDark: '#f6f9fc',
  backgroundLight: '#fff',
  border: '#666666',
  text: '#333333'
};

export const getUiTheme = theme => {
  if (theme === undefined) return darkTheme;

  if (typeof theme === 'string') {
    if (theme === 'dark') {
      return darkTheme;
    } else if (theme === 'light') {
      return lightTheme;
    }

    throw Error(
      `Theme "${theme}" is not available. Please use "light" or "dark", or provide a custom theme object.`
    );
  }

  if (theme.base && theme.base === 'light') {
    return {
      ...lightTheme,
      ...theme
    };
  }

  return {
    ...darkTheme,
    ...theme
  };
};
