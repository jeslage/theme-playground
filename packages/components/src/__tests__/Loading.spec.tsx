import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import Loading from '../components/Loading/Loading';
import StyledLoading from '../components/Loading/Loading.style';
import { darkTheme } from '../theme';

describe('Loading', () => {
  it('Loading should render correctly', () => {
    const component = shallow(<Loading isLoading />);

    expect(component).toMatchSnapshot();
  });

  it('StyledLoading should render correctly', () => {
    const component = shallow(
      <StyledLoading theme={{ colors: darkTheme }} isLoading={true} />
    );

    expect(component).toMatchSnapshot();
  });
});
