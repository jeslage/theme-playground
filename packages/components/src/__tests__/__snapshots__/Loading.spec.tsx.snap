// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Loading should render correctly 1`] = `
<styled.div
  isLoading={true}
>
  <div />
  <span>
    Updating
  </span>
</styled.div>
`;

exports[`Loading StyledLoading should render correctly 1`] = `
<div
  className="sc-AxjAm deQmBI"
/>
`;
